### PR TITLE
 [ioctl] allow get block number 0 (genesis block)

### PR DESCRIPTION
**Pull request by dustinxie**
also added tx dump, in case the failed tx happen again, the protobuf bytes are dumped so we can know the root-cause

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -364,7 +364,12 @@ func (api *Server) SendAction(ctx context.Context, in *iotexapi.SendActionReques
 	hash := selp.Hash()
 	l := log.L().With(zap.String("actionHash", hex.EncodeToString(hash[:])))
 	if err = api.ap.Add(ctx, selp); err != nil {
-		l.Error("Failed to accept action", zap.Error(err))
+		txBytes, serErr := proto.Marshal(in.Action)
+		if serErr != nil {
+			l.Error("Data corruption", zap.Error(serErr))
+		} else {
+			l.With(zap.String("txBytes", hex.EncodeToString(txBytes))).Error("Failed to accept action", zap.Error(err))
+		}
```
Comment (dustinxie) [inline api/api.go:372]: dumped failed tx bytes

File: ioctl/cmd/bc/bcblock.go [hunk 2]
```diff
@@ -129,15 +128,16 @@ func (m *blockMessage) String() string {
...
+		blocksInfos []blocksInfo
+	)
 	isHeight := true
 	if len(args) != 0 {
-		height, err = strconv.ParseUint(args[0], 10, 64)
-		if err != nil {
+		if height, err = strconv.ParseUint(args[0], 10, 64); err != nil {
 			isHeight = false
-		} else if err = validator.ValidatePositiveNumber(int64(height)); err != nil {
-			return output.NewError(output.ValidationError, "invalid height", err)
```
Comment (dustinxie) [inline ioctl/cmd/bc/bcblock.go:140]: allow to get block height = 0

File: api/api.go [hunk 3]
```diff
@@ -1241,9 +1248,11 @@ func (api *Server) getBlockMetaByHeight(height uint64) (*iotextypes.BlockMeta, e
...
 	}
 	// get block's receipt
-	blk.Receipts, err = api.dao.GetReceipts(height)
-	if err != nil {
-		return nil, status.Error(codes.NotFound, err.Error())
+	if blk.Height() > 0 {
+		blk.Receipts, err = api.dao.GetReceipts(height)
+		if err != nil {
+			return nil, status.Error(codes.NotFound, err.Error())
+		}
```
Comment (dustinxie) [inline api/api.go:1255]: genesis block does not have receipt

File: api/api.go [hunk 4]
```diff
@@ -1254,7 +1263,10 @@ func generateBlockMeta(blk *block.Block) *iotextypes.BlockMeta {
 	hash := header.HashBlock()
 	height := header.Height()
 	ts, _ := ptypes.TimestampProto(header.Timestamp())
-	producerAddress := header.ProducerAddress()
+	var producerAddress string
+	if blk.Height() > 0 {
+		producerAddress = header.ProducerAddress()
+	}
```
Comment (dustinxie) [inline api/api.go:1269]: genesis block does not have producer

**Review by raullenchai — APPROVED** [review]

**Pull request merged**